src/plugins/routing.js

```diff
@@ -1,19 +1,25 @@
 import './middleware'
 import Vue from 'vue'
 
+const STRATEGIES = <%= JSON.stringify(options.STRATEGIES) %>
+const STRATEGY = '<%= options.strategy %>'
 const vuex = <%= JSON.stringify(options.vuex) %>
 const routesNameSeparator = '<%= options.routesNameSeparator %>'
+const defaultLocale = '<%= options.defaultLocale %>'
 
 function localePathFactory (i18nPath, routerPath) {
-  const STRATEGIES = <%= JSON.stringify(options.STRATEGIES) %>
-  const STRATEGY = '<%= options.strategy %>'
-  const defaultLocale = '<%= options.defaultLocale %>'
   const defaultLocaleRouteNameSuffix = '<%= options.defaultLocaleRouteNameSuffix %>'
 
   return function localePath (route, locale) {
     // Abort if no route or no locale
     if (!route) return
+
+    if (STRATEGY === STRATEGIES.NO_PREFIX && locale !== this.$i18n.locale) {
```

Comment:
Should also have `&& locale` condition so that it doesn't warn when not passing locale.

Reply:
Missed this before: should also use `this[i18nPath].locale`.